test/service/test_response.py

```diff
@@ -1588,13 +1588,25 @@ def tearDownClass(cls):
         cls._teardown_indices()
         super().tearDownClass()
 
+    def _count_total_cells(self, response_json):
+        """
+        Return the number of cell suspension inner entities and total cell count
+        per hit
+        """
+        return [
+            (
+                len(hit['cellSuspensions']),
+                sum([cs['totalCells'] for cs in hit['cellSuspensions']])
+            )
+            for hit in response_json['hits']
+        ]
+
     def test_sorting_by_cell_count(self):
         """
-        Verify sorting by 'cellCount' sorts each hit by the sum of 'totalCells'
+        Verify sorting by 'cellCount' considers the sum of 'totalCells'
```

Comment:
Nitpick: isn't `totalCells` already a sum? If that's so them I think
```suggestion
        Verify sorting by 'cellCount' considers the value of 'totalCells'
```

Reply:
> This looks good! Nice work. How come you had to add a post-aggregate step? Why can't the cell count be properly calculated during aggregation?

When the aggregation is performed in `IndexService._aggregate_entity()` it is done per `entity_type`:
```
for entity_type, entities in contents.items():
    ...
    aggregate_contents[entity_type] = entities
```
The post-aggregate step is required to inject the new (and HCA specific) property `total_estimated_cells` at the `entity_type` level (eg. `contents.total_estimated_cells`). Doing it this way allows the `IndexService` to remain generic by putting the HCA specific transformation code in the HCA metadata plugin's transformer.

Reply:
> Nitpick: isn't `totalCells` already a sum? If that's so then I think

It is true that `totalCells` is a summed value in that `cell_suspension` inner entities are grouped by organ, however when filtering and sorting by `totalCells` the value to be considered is the sum of the `totalCells` property across all the groups. I'll modify the docstring to help clarify.